clean/scott_pd_pprr.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+from lib.columns import clean_column_names, set_values
+from lib.path import data_file_path, ensure_data_dir
+from lib.uid import gen_uid
+from lib import salary
+from lib.clean import (
+    clean_names, clean_dates, clean_sexes, clean_races, clean_salaries
+)
+import sys
+sys.path.append("../")
+
+
+def split_name(df):
+    series = df.full_name.fillna("").str.strip()
+    for col, pat in [("first_name", r"^([\w'-]+)(.*)$"), ("middle_initial", r"^(\w\.) (.*)$")]:
+        names = series[series.str.match(pat)].str.extract(pat)
+        df.loc[series.str.match(pat), col] = names[0]
+        series = series.str.replace(pat, r"\2").str.strip()
+    df.loc[:, "last_name"] = series
+    return df.drop(columns=["full_name"])
```

Comment:
@ayyubibrahimi try this:
```python
df.full_name.str.strip().str.lower().str.extract(r'^([^ ]+) (?:([^ ]+) )?(.+)')
```

Reply:
Regex skill is very valuable. I recommend reading the [Mastering Regex](https://www.amazon.com/Mastering-Regular-Expressions-Understand-Productive-ebook/dp/B007I8S1X0/ref=tmm_kin_swatch_0?_encoding=UTF8&qid=&sr=) book. You will find that overtime the payback is far far greater than this little effort.

Reply:
Just got the book. Thanks!

Reply:
Didn't mean to push to main. Kept getting error messages as I pushed to scott and somehow slipped up and pushed to main accidently. Looking for ways to undo it now.

Reply:
I was able to correct the issue in the "middle_initial" column with: ("middle_initial", r"^(\w\.|\w\s)(.*)$")